src/runtime/virtcontainers/utils/utils.go

```diff
@@ -375,8 +375,19 @@ outer:
 	if pidRunning {
 		// Force process to die
 		if err = syscall.Kill(pid, syscall.SIGKILL); err != nil {
+			if err == syscall.ESRCH {
+				logger.WithField("pid", pid).Warnf("kill encounters ESRCH, process already finished")
+				return nil
+			}
 			return fmt.Errorf("Failed to stop process %v: %s", pid, err)
 		}
+
+		for {
+			_, err := syscall.Wait4(pid, nil, 0, nil)
```

Comment:
The existing code base uses `syscall.Wait4`. Not sure `syscall.Waitpid` is available everywhere, e.g. Darwin.

Reply:
Waitpid/SYS_WAITPID is not defined for Darwin in syscall or golang.org/x/sys/unix. syscall.Wait4 is the function used internally in go runtime library.

Reply:
Yes. Ideally we'd want to rely on the go runtime library `Cmd.Wait()` or `Process.Wait()` to handle such platform details but this would require more changes that go beyond the goal of this PR.

Reply:
OK. So let's keep it as is. I'm surprised you only have the non-POSIX syscall.